pkgs/applications/editors/emacs/elisp-packages/tsc/default.nix

```
@@ -0,0 +1,73 @@
+{ lib
+, symlinkJoin
+, melpaBuild
+, fetchFromGitHub
+, rustPlatform
+, writeText
+, clang
+, llvmPackages
+}:
+with lib.licenses;
+with rustPlatform;
+with llvmPackages;
+
+let
+  version = "0.16.1";
+
+  src = fetchFromGitHub {
+    owner = "emacs-tree-sitter";
+    repo = "elisp-tree-sitter";
+    rev = version;
+    sha256 = "sha256-tAohHdAsy/HTFFPSNOo0UyrdolH8h0KF2ekFXuLltBE=";
+  };
+
+  tsc = melpaBuild rec {
```

nit: `rec` not needed
```suggestion
  tsc = melpaBuild {
```